resources/task.rb

```diff
@@ -0,0 +1,149 @@
+property :task_name, String, desired_state: false, identity: true, name_attribute: true
+property :task_source, String, desired_state: false, identity: true, default: ''.freeze
+property :task_crontab, String, desired_state: false, identity: true, default: '0 1 * * * ?'.freeze
+property :api_endpoint, String, desired_state: false, identity: true,
+                                default: lazy { node['nexus3']['api']['endpoint'] }
+property :api_username, String, desired_state: false, identity: true,
+                                default: lazy { node['nexus3']['api']['username'] }
+property :api_password, String, desired_state: false, identity: true, sensitive: true,
+                                default: lazy { node['nexus3']['api']['password'] }
+
+load_current_value do |desired|
+  api_endpoint desired.api_endpoint
+  api_username desired.api_username
+  api_password desired.api_password
+
+  def apiclient
+    @apiclient ||= ::Nexus3::Api.new(api_endpoint, api_username, api_password)
+  end
+
+  begin
+    res = apiclient.run_script('get_task', desired.task_name)
+    current_value_does_not_exist! if res == 'null'
+    config = JSON.parse(res)
+    ::Chef::Log.warn "Config is: #{config}"
+    task_name config['.name']
+    task_source config['source']
+  # We rescue here because during the first run, the task will not exist yet, so we let Chef know that
+  # the resource has to be created.
+  rescue LoadError, ::Nexus3::ApiError => e
+    ::Chef::Log.warn "A '#{e.class}' occured: #{e.message}"
+    current_value_does_not_exist!
+  end
+end
+
+action :create do
+  init
+
+  converge_if_changed do
+    nexus3_api "upsert_task #{new_resource.task_name}" do
+      script_name 'upsert_task'
+      args name: new_resource.task_name,
+           source: new_resource.task_source,
+           crontab: new_resource.task_crontab
+
+      action %i(create run)
+      endpoint new_resource.api_endpoint
+      username new_resource.api_username
+      password new_resource.api_password
+
+      content <<-EOS
```

Comment:
Move the content to the file directory and use `::Nexus3::Scripts.groovy_content('upsert_task', node)`

Reply:
done